src/Tests/Microsoft.NET.Build.Tests/KnownRuntimePackTests.cs

```diff
@@ -147,7 +147,7 @@ public void ItCanPublishArm64Winforms()
             newCommand.Execute().Should().Pass();
 
             new PublishCommand(Log, testDirectory)
-                .Execute("/p:RuntimeIdentifier=win-arm64")
+                .Execute("/p:RuntimeIdentifier=win-arm64", "/p:TargetPlatformIdentifier=Windows", "/p:TargetPlatformVersion=7.0")
```

Comment:
Until we have the TargetPlatform support in the TargetFramework, I think we should add logic that will set the TargetPlatformIdentifier to Windows if UseWindowsForms or UseWPF is true.  Then we won't break those projects, requiring them to set the TargetPlatformIdentifier explicitly, when what we want them to do eventually is to update the TargetFramework to include `-windows`.

Reply:
I've updated so the target platform properties are defined with UseWPF or UseWindowsForms is set. Note that we also discussed this change over email in relation to #12418 and the preview 7 XAML designer bug.